perf/src/packet.rs

```diff
@@ -137,6 +137,20 @@ where
         .deserialize_from(data)
 }
 
+pub fn deserialize_from_with_limit<R, T>(reader: R) -> bincode::Result<T>
+where
+    R: Read,
+    T: DeserializeOwned,
+{
+    // with_limit causes pre-allocation size to be limited
+    // to prevent against memory exhaustion attacks.
+    bincode::options()
+        .with_limit(PACKET_DATA_SIZE as u64)
```

Comment:
(Y)

Reply:
?

Reply:
I thought that will result in a thumb-up emoji, but it seems not the case in the github.